src/clj/game/core/runs.clj

```diff
@@ -234,39 +238,41 @@
         choices (into [] (concat ability-strs steal-str no-action-str))]
     ;; Steal costs are additional costs and can be denied by the runner.
     (continue-ability state :runner
-                      (if (> (count cost-strs) 1)
-                        ;; Use the better function for multiple costs
-                        (steal-pay-choice state :runner cost-strs '() n c)
-                        ;; Otherwise, just handle everything right friggin here
-                        {:delayed-completion true
-                         :prompt prompt-str
-                         :choices choices
-                         :effect (req (cond
-                                        ;; Can't steal or pay, or won't pay single additional cost to steal
-                                        (= target "No action")
-                                        (access-end state side eid c)
-
-                                        ;; Steal normally
-                                        (= target "Steal")
-                                        (steal-agenda state :runner eid c)
-
-                                        ;; Pay single additiional cost to steal
-                                        (.contains target "Pay")
+                      {:delayed-completion true
+                       :prompt prompt-str
+                       :choices choices
+                       :effect (req (cond
+                                      ;; Can't steal or pay, or won't pay single additional cost to steal
+                                      (= target "No action")
+                                      (access-end state side eid c)
+
+                                      ;; Steal normally
+                                      (= target "Steal")
+                                      (steal-agenda state :runner eid c)
+
+                                      ;; Pay single additiional cost to steal
+                                      (.contains target "Pay")
+                                      (if (> n 1)
+                                        ;; Use the better function for multiple costs
+                                        (resolve-ability state :runner
```

Comment:
I'm fairly certain this needs to be continue-ability. Otherwise this eid is left hanging.

Reply:
Damn, good call. Thanks for catching that.